crates/core/src/layout/store.rs

```diff
@@ -31,6 +32,16 @@ pub struct Store {
 impl Store {
     #[instrument(name = "create_store")]
     pub fn new(dir: &Path) -> Self {
+        // Allow overriding the temp/download cache directory via PROTO_DOWNLOAD_CACHE
+        let temp_dir = get_download_cache_dir().unwrap_or_else(|| dir.join("temp"));
```

Comment:
Maybe it should just be `PROTO_TEMP_DIR` and used directly? Not sure the nested `temp` is necessary.